src/search.rs

```diff
@@ -0,0 +1,755 @@
+                ValueKind::Toughness => "toughness",
+                ValueKind::PowTou => "powtou",
+                ValueKind::Loyalty => "loyalty",
+                ValueKind::Rarity => "rarity",
+                ValueKind::Set => "set",
+                ValueKind::Number => "number",
+                ValueKind::Block => "block",
+                ValueKind::SetType => "settype",
+                ValueKind::Cube => "cube",
+                ValueKind::Format => "format",
+                ValueKind::Banned => "banned",
+                ValueKind::Restricted => "restricted",
+                ValueKind::Usd => "usd",
+                ValueKind::UsdFoil => "usdfoil",
+                ValueKind::Eur => "eur",
+                ValueKind::Tix => "tix",
+                ValueKind::Cheapest => "cheapest",
+                ValueKind::Artist => "artist",
+                ValueKind::ArtistCount => "artists",
+                ValueKind::Flavor => "flavor",
+                ValueKind::Watermark => "watermark",
+                ValueKind::IllustrationCount => "illustrations",
+                ValueKind::BorderColor => "border",
+                ValueKind::Frame => "frame",
+                ValueKind::Year => "year",
+                ValueKind::Date => "date",
+                ValueKind::PrintCount => "prints",
+                ValueKind::SetCount => "sets",
+                ValueKind::PaperPrintCount => "paperprints",
+                ValueKind::PaperSetCount => "papersets",
+                ValueKind::Game => "game",
+                ValueKind::Language => "language",
+                ValueKind::InRarity
+                | ValueKind::InSet
+                | ValueKind::InSetType
+                | ValueKind::InGame
+                | ValueKind::InLanguage => "in",
+                ValueKind::Name => "name",
+                // TODO(msmorgan): Should "Exact" be a ValueKind?
+                ValueKind::Exact => "",
+            }
+        )
+    }
+}
+
+pub enum Guild {
+    Azorius,
+    Boros,
+    Dimir,
+    Golgari,
+    Gruul,
+    Izzet,
+    Orzhov,
+    Rakdos,
+    Selesnya,
+    Simic,
+}
+
+pub enum Shard {
+    Bant,
+    Esper,
+    Grixis,
+    Jund,
+    Naya,
+}
+
+pub enum Wedge {
+    Abzan,
+    Jeskai,
+    Mardu,
+    Sultai,
+    Temur,
+}
+
+pub enum FourColor {
+    Aggression,
+    Altruism,
+    Artifice,
+    Chaos,
+    Growth,
+}
+
+#[derive(Clone, Eq, PartialEq, Hash, Debug)]
+pub enum CompareOp {
+    Lte,
+    Lt,
+    Gte,
+    Gt,
+    Eq,
+    Neq,
+}
+
+impl fmt::Display for CompareOp {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str(match self {
+            CompareOp::Lte => "<=",
+            CompareOp::Lt => "<",
+            CompareOp::Gte => ">=",
+            CompareOp::Gt => ">",
+            CompareOp::Eq => "=",
+            CompareOp::Neq => "!=",
+        })
+    }
+}
+
+#[derive(Clone, Eq, PartialEq, Hash, Debug)]
+pub struct Compare<T> {
+    op: Option<CompareOp>,
```

Comment:
why is the compare operation optional?

Reply:
Otherwise it's a colon.

Reply:
I don't get ti :thinking:

Reply:
I had two other choices. I could make the `Param::Value` look like `Param::Value(ValueKind, Option<CompareOp>, Box<...>)` (where `None` is a colon, such as in the param `name:"stormcrow"`), or I could have added a third variant `Param::Compare(ValueKind, CompareOp, Box<...>)`. I wanted to be able to write `cmc(gt(3))` and such,  so I figured I would use a structure for comparisons, and then convert bare values to compare without an `op` (i.e., a colon).

The Scryfall API might treat : and = equally. With some testing I could remove the colon case entirely if that turns out to be true.

Reply:
> The Scryfall API might treat : and = equally. With some testing I could remove the colon case entirely if that turns out to be true.

No, Scryfall does not treat them equally.
Example: 
https://scryfall.com/search?q=id%3Aubg&unique=cards&as=grid&order=name
vs
https://scryfall.com/search?q=id%3Dubg&unique=cards&as=grid&order=name

Reply:
@jheyens Thank you! I just discovered this myself. It seems like `:` is equivalent to `>=` when it comes to colors.